web/src/components/tasks/NaicsCodeTask.test.tsx

```diff
@@ -417,14 +417,53 @@ describe("<NaicsCodeTask />", () => {
       expect(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`)).toBeInTheDocument();
     });
 
-    it("opens Needs Account modal on save button click", async () => {
+    it("opens Needs Account modal on NAICS Code radio button click", async () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
       renderPage();
-      fireEvent.change(screen.getByLabelText("Save NAICS Code"), {
-        target: { value: validNaicsCode },
+
+      fireEvent.click(screen.getByTestId("naics-radio-621399"));
+
+      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
+      await waitFor(() => {
+        expect(screen.getByTestId("naics-radio-621399")).not.toBeChecked();
       });
-      fireEvent.click(screen.getByText(`Register & ${Config.determineNaicsCode.saveButtonText}`));
+    });
+
+    it("opens Needs Account modal on NAICS Code input radio button click", async () => {
+      initialBusiness = generateBusiness({
+        profileData: generateProfileData({ naicsCode: "", industryId: "acupuncture" }),
+      });
+      renderPage();
+
+      fireEvent.click(screen.getByTestId("naics-radio-input"));
+
+      expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
       await waitFor(() => {
-        return expect(setShowNeedsAccountModal).toHaveBeenCalledWith(true);
+        expect(screen.getByTestId("naics-radio-input")).not.toBeChecked();
```

Comment:
Same as the above comment regarding this never failing.

Here is the correct line of code that needs to be used here. Note that the waitFor is unnecessary and should be removed.
```
expect(within(screen.getByTestId("naics-radio-input")).getByRole("radio")).not.toBeChecked();
```